director/stat.go

```diff
@@ -345,15 +360,24 @@ func (stat *ObjectStat) queryServersForObject(ctx context.Context, objectName st
 		}
 		// Use an anonymous func to pass variable safely to the goroutine
 		func(serverAd server_structs.ServerAd) {
-			statUtil.Errgroup.Go(func() error {
-				baseUrl := serverAd.URL
-
-				// For the topology server, if the server does not support public read,
-				// or the token is provided, or the object is protected, then it's safe to assume this request goes to authenticated endpoint
-				// For Pelican server, we don't populate authURL and only use server URL as the base URL
-				if serverAd.FromTopology && (!serverAd.Caps.PublicReads || cfg.protected || cfg.token != "") && serverAd.AuthURL.String() != "" {
-					baseUrl = serverAd.AuthURL
-				}
+
+			baseUrl := serverAd.URL
+			// For the topology server, if the server does not support public read,
+			// or the token is provided, or the object is protected, then it's safe to assume this request goes to authenticated endpoint
+			// For Pelican server, we don't populate authURL and only use server URL as the base URL
+			if serverAd.FromTopology && (!serverAd.Caps.PublicReads || cfg.protected || cfg.token != "") && serverAd.AuthURL.String() != "" {
+				baseUrl = serverAd.AuthURL
```

Comment:
In my testing, I observed that a tokenless stat request for `/chtc/PUBLIC/matyas/robots.txt` wound up being denied at every topology cache because their representative server ads had "Caps.PublicReads = false". This doesn't make sense for caches, and it doesn't make sense to send a tokenless stat requests to a cache's auth URL. I think the issue is in the way we parse topology ads. Unfortunately, the `serverAd` struct doesn't have a way to encode whether the server is a cache or an origin, so we can't directly check that here. One temporary solution I identified is to modify `director/advertise.go::parseServerAdFromTopology:65` to set the PublicReads capability to true. A quick test of that locally bumped the number of caches found to have the object from 1 to 8.